apps/main/src/entities/withdraw/api/withdraw-query.ts

```diff
@@ -0,0 +1,219 @@
+import type { QueryFunction } from '@tanstack/react-query'
+import type {
+  Amount,
+  ClaimableDetailsResp,
+  WithdrawDetailsResp,
+  WithdrawEstGasApprovalResp,
+  WithdrawQueryKeyType,
+} from '@/entities/withdraw'
+
+import { isBonus, isHighSlippage, isValidAddress } from '@/utils'
+import { warnIncorrectEstGas } from '@/lib/curvejs'
+import useStore from '@/store/useStore'
+
+export const withdrawDetails: QueryFunction<WithdrawDetailsResp, WithdrawQueryKeyType<'withdrawDetails'>> = async ({
+  queryKey,
+}) => {
+  const [, , poolId, selected, lpToken, formAmounts, selectedTokenAddress, isWrapped, maxSlippage] = queryKey
```

Comment:
Follow the good naming. Query shouldn't know where params came from.
```suggestion
  const [, , poolId, type, lpToken, amounts, token, isWrapped, maxSlippage] = queryKey
```